src/adapter/cloudflare-pages/handler.ts

```diff
@@ -34,6 +43,54 @@ export const handle: HandleInterface = (app: Hono) => (eventContext: EventContex
   )
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function handleMiddleware<E extends Env = any, P extends string = any, I extends Input = {}>(
+  middleware: MiddlewareHandler<E, P, I>
+): PagesFunction {
+  return async (executionCtx) => {
+    const context = new Context(executionCtx.request, {
+      env: executionCtx.env,
+      executionCtx,
+    })
+
+    let response: Response | void = undefined
+
+    try {
+      response = await middleware(context, async () => {
+        try {
+          context.res = await executionCtx.next()
+        } catch (error) {
+          if (error instanceof Error) {
+            context.error = error
+          } else {
+            throw error
+          }
+        }
+      })
+    } catch (error) {
+      if (error instanceof Error) {
+        context.error = error
+      } else {
+        throw error
+      }
+    }
+
+    if (response) {
+      return response
+    }
+
+    if (context.error instanceof HTTPException) {
+      return context.error.getResponse()
+    }
+
+    if (context.error) {
+      throw context.error
+    }
+
+    return context.res
+  }
```

Comment:
I'm still not sure all of this is necessary.. I added all the test cases I could think of, so hopefully all the main use cases are covered